tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/CreatePurchasingDocumentServiceTests.cs

```diff
@@ -13,32 +14,61 @@ namespace NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Services
     [Parallelizable(ParallelScope.All)]
     internal sealed class CreatePurchasingDocumentServiceTests
     {
+        [Test]
+        public void Constructor_NullParameter_ThrowsArgumentNullException()
+        {
+            var builder = CreatePurchasingDocumentServiceBuilder
+                .Create()
+                .WithPatientNumbersCsvWriter(null);
+
+            Assert.Throws<ArgumentNullException>(() => builder.Build());
+        }
+
         [TestCase(true, false)]
         [TestCase(false, true)]
         [TestCase(false, false)]
-        public void CreateDocumentAsync_ArgumentsAreNull_ReturnNullArgumentException(bool hasStream, bool hasOrder)
+        public void CreatePatientNumbersCsvAsync_ArgumentsAreNull_ReturnNullArgumentException(bool hasStream, bool hasOrder)
         {
-            var purchasingDocumentService = new CreatePurchasingDocumentService();
+            var context = CreatePurchasingDocumentServiceTestContext.Setup();
 
             Assert.ThrowsAsync<ArgumentNullException>(() =>
-                purchasingDocumentService.CreateDocumentAsync(
+                context.CreatePurchasingDocumentService.CreatePatientNumbersCsvAsync(
                     hasStream ? new MemoryStream() : null,
                     hasOrder ? OrderBuilder.Create().Build() : null));
         }
 
         [Test]
-        public async Task CreateDocumentAsync_DocumentIsCreated_StreamIsAsExpected()
+        public async Task CreatePatientNumbersCsvAsync_DocumentIsCreated_StreamIsAsExpected()
```

Comment:
Name of this test is unexpected. Am I right in thinking rather than checking the contents of the stream it now ensuring the AttachmentCsvWriter is called once?